lib/tasks/update_resources.rake

```diff
@@ -0,0 +1,99 @@
+desc "This task will update resources in the database from the webcheckout api"
+task update_resources: :environment do
+  host = "https://webcheckout.lsa.umich.edu"
+  userid = Rails.application.credentials.dig(:webcheckout_api, :userid)
+  password = Rails.application.credentials.dig(:webcheckout_api, :password)
+
+  begin
+    rooms = Room.all.pluck(:rmrecnbr)
+
+    #=============================
+    #   get oids from api        #
+    #=============================
+    api = WebcheckoutApi.new(host, userid, password)
+    api.start_session
+    payload = api.get_location_oids(rooms)
+    location_oids = payload['result'].map { |item| item['oid'].to_i }
+    room_location = payload['result'].map { |item| [item['oid'], item['barcode']] }.to_h
+    api.end_session
+
+    #=============================
+    #   get resources from api   #
+    #=============================
+    api = WebcheckoutApi.new(host, userid, password)
+    api.start_session
+    wco_resources = location_oids.flat_map do |oid|
+      payload = api.get_resources(oid)
+      if payload['count'] > 0
+        payload['result'].map { |resource| [oid, resource['name'], resource['resourceType']['name']] }
+      else
+        []
+      end
+    end
+    api.end_session
+
+    #=============================
+    #   update the database      #
+    #=============================
+    type_names = ['Wireless Bodypack Transmitter', "Instructional PC Desktop",
+                  "Blu-Ray Player", "Lecture Capture Camera", "Video Projector"]
```

Comment:
Let's get type_names from AppPreferences. 
On Admin->Create Preferences page, create a Preference:
```
name: "resource_types",
description: " Comma-separated list of resource types to add from WebCheckout",
pref_type: "string"
```
On Admin->App Preferences add the following list to that preference:
```
"Wireless Bodypack Transmitter,  Instructional PC Desktop,  Lecture Capture Camera,  Video Projector" 
```
In the code:
```
type_names = AppPreference.find_by(name: "resource_types").value.split(",").each(&:strip!)
```